File: halcyon-web/src/main/resources/static/js/controller/hr/hrCompanyController.js
//控制层
app.controller("hrCompanyController", function ($scope, $controller, hrCompanyService) {
    //继承
    $controller("baseController", {$scope: $scope});

    // $scope.hrCompany = {};
    $scope.query = function () {
        hrCompanyService.query().success(function (responseData) {
            if (responseData.length > 0) {
                $scope.hrCompany = responseData[0];
            } else {
                $scope.hrCompany = {};
            }
        });
    }

    $scope.save = function () {
        hrCompanyService.save($scope.hrCompany).success(function (responseData) {
            $scope.parseResponse(responseData);
            $scope.query();
        });
    }

});

File: halcyon-base/src/main/java/com/marshal/halcyon/base/hr/controller/HrCompanyController.java
package com.marshal.halcyon.base.hr.controller;

import com.marshal.halcyon.core.entity.ResponseData;
import com.marshal.halcyon.core.controller.BaseController;
import com.marshal.halcyon.base.hr.entity.HrCompany;
import com.marshal.halcyon.base.hr.service.HrCompanyService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * @auth: Marshal
 * @date: 2019/1/9
 * @desc:
 */
@RestController
@RequestMapping("/hr/company")
public class HrCompanyController extends BaseController {

    @Autowired
    HrCompanyService hrCompanyService;

    @RequestMapping("/query")
    public List<HrCompany> query() {
        return hrCompanyService.selectAll();
    }

    @RequestMapping("/save")
    public ResponseData save(@RequestBody HrCompany hrCompany) {
        hrCompanyService.save(hrCompany);
        return new ResponseData();
    }
}
